app/controllers/modules/course/courses_controller/announcements_module.rb

```diff
@@ -0,0 +1,12 @@
+class Course::CoursesController::AnnouncementsModule
```

Comment:
I think this naming doesn't make sense... Why not just call it `Course::AnnouncementsModule`